sphinxext/rediraffe.py

```diff
@@ -211,10 +211,9 @@ def build_redirects(app: Sphinx, exception: Union[Exception, None]) -> None:
         redirect_to = src_redirect_to.with_suffix("")
 
         if type(app.builder) == DirectoryHTMLBuilder:
-            master_doc = Path(app.config.master_doc).with_suffix("")
-            if redirect_from != master_doc:
+            if redirect_from.name != "index":
                 redirect_from = redirect_from / "index"
-            if redirect_to != master_doc:
+            if redirect_to.name != "index":
```

Comment:
>   if redirect_to.name != "index":

Just a note that this may break in the future. Anyways, the default in Sphinx 2+ is index, so we don't break *that* many.